pennylane/devices/tests/test_templates.py

```diff
@@ -225,6 +225,15 @@ def circuit():
             [math.fidelity_statevector(circuit(), exp_state)], [1.0], atol=tol(dev.shots)
         )
 
+    @pytest.fixture(scope="function", autouse=True)
+    def capture_warnings(self, recwarn):
+        """Capture warnings."""
+        yield
+        if len(recwarn) > 0:
+            for w in recwarn:
+                assert isinstance(w.message, qml.PennyLaneDeprecationWarning)
+                assert "BasisStatePreparation is deprecated" in str(w.message)
```

Comment:
This fixture captures all warnings in all tests and assert that they are raised from `BasisStatePreparation`, including warnings raised from deprecated devices. Maybe instead of a fixture, use `with pytest.raises` in the individual tests that uses the deprecated features?

Reply:
The error seems to pop up in all tests regardless of whether they use BasisStatePreparation or not. So I don't think the problem is in the decorator 🤔 Maybe this test is called from somewhere that instantiates the template?

Reply:
or maybe you are right 😆